polyfactory/factories/base.py

```diff
@@ -863,25 +893,41 @@ def process_kwargs(cls, **kwargs: Any) -> dict[str, Any]:
                     result[field_meta.name] = cls._handle_factory_field(
                         field_value=field_value,
                         field_build_parameters=field_build_parameters,
+                        build_context=_build_context,
                     )
                     continue
 
-                result[field_meta.name] = cls.get_field_value(field_meta, field_build_parameters=field_build_parameters)
+                field_result = cls.get_field_value(
+                    field_meta,
+                    field_build_parameters=field_build_parameters,
+                    build_context=_build_context,
+                )
+                if field_result is Null:
+                    continue
+
+                result[field_meta.name] = field_result
 
         for field_name, post_generator in generate_post.items():
             result[field_name] = post_generator.to_value(field_name, result)
 
         return result
 
     @classmethod
-    def process_kwargs_coverage(cls, **kwargs: Any) -> abc.Iterable[dict[str, Any]]:
+    def process_kwargs_coverage(
+        cls,
+        _build_context: BuildContext | None = None,
+        **kwargs: Any,
+    ) -> abc.Iterable[dict[str, Any]]:
         """Process the given kwargs and generate values for the factory's model.
 
         :param kwargs: Any build kwargs.
 
         :returns: A dictionary of build results.
 
         """
+        _build_context = _get_build_context(_build_context)
+        _build_context["seen_models"].add(cls.__model__)
+
```

Comment:
This doesn't seem to be used anywhere.

Reply:
Good catch. This should be passed to downstream methods. I'll update